// Proto file for connection between the client and intermediate balancer, for registering end machines at the intermediate, for scaling up and down
// python3 -m grpc_tools.protoc -I. --python_out=. --pyi_out=. --grpc_python_out=. trial_1.proto

syntax = "proto3";
package trial_1;

// To store the message from the client
message FunctionMessage
{
    optional int32 data1 = 1;
    optional int32 data2 = 2;
    optional int32 function = 3;
}

message ReturnValue
{
    optional float val = 1;
}

// Message type for end machines to register themselves at the intermediate
message MachineInfo
{
    optional string ip = 1;
}

// Message type for scaling up or scaling down at each machine
message Scale
{
    optional int32 type = 1;
}

// Void message for when nothing actually needs to be returned
message void {}

service Alert
{
    rpc InvokeMethod(FunctionMessage) returns (ReturnValue) {}
    rpc RegisterMachine(MachineInfo) returns (void) {}
    rpc ChangeScale(Scale) returns (void) {}
}
